samcli/commands/remote_invoke/remote_invoke_context.py

```diff
@@ -0,0 +1,122 @@
+"""
+Context object used by `sam remote invoke` command
+"""
+import logging
+from typing import Optional, cast
+
+from samcli.commands.remote_invoke.exceptions import (
+    AmbiguousResourceForRemoteInvoke,
+    InvalidRemoteInvokeParameters,
+    NoResourceFoundForRemoteInvoke,
+    UnsupportedServiceForRemoteInvoke,
+)
+from samcli.lib.pipeline.bootstrap.resource import ARNParts
+from samcli.lib.utils.boto_utils import BotoProviderType
+from samcli.lib.utils.cloudformation import (
+    CloudFormationResourceSummary,
+    get_resource_summaries,
+    get_resource_summary_from_physical_id,
+)
+from samcli.lib.utils.resources import AWS_LAMBDA_FUNCTION
+
+LOG = logging.getLogger(__name__)
+
+
+SUPPORTED_SERVICES = {"lambda": AWS_LAMBDA_FUNCTION}
+
+
+class RemoteInvokeContext:
+
+    _boto_client_provider: BotoProviderType
+    _boto_resource_provider: BotoProviderType
+    _stack_name: Optional[str]
+    _resource_id: Optional[str]
+    _resource_summary: Optional[CloudFormationResourceSummary]
+
+    def __init__(
+        self,
+        boto_client_provider: BotoProviderType,
+        boto_resource_provider: BotoProviderType,
+        stack_name: Optional[str],
+        resource_id: Optional[str],
+    ):
+        self._boto_resource_provider = boto_resource_provider
+        self._boto_client_provider = boto_client_provider
+        self._stack_name = stack_name
+        self._resource_id = resource_id
+        self._resource_summary = None
+
+    def __enter__(self) -> "RemoteInvokeContext":
+        self._populate_resource_summary()
+        return self
+
+    def __exit__(self, *args) -> None:
+        pass
+
+    def run(self):
+        pass
+
+    def _populate_resource_summary(self) -> None:
+        if not self._stack_name and not self._resource_id:
+            raise InvalidRemoteInvokeParameters("Either --stack-name or --resource-id parameter should be provided")
```

Comment:
🤔 agreed on this one, this validation can be moved up to `command.py` through `click`. @hnnasit what do you think?

Reply:
I think that makes sense. We will also have another validation for `payload` and `payload-file` in command.py. We can move the above validation to cli as well.

Reply:
Regarding to this one, I feel weird when removing this validation since typing system can't guarantee one or the other field should have a value. After discussing with @hnnasit we decided to keep it here as well as adding it to the `command.py` implementation later.